sale_subscription_financial_risk/models/sale_subscription.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (C) 2019 Amplex
+# Copyright (C) 2019 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from datetime import date, timedelta
+from dateutil.relativedelta import relativedelta
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class SaleSubscription(models.Model):
+    _inherit = 'sale.subscription'
+
+    # Calculates and returns a timedelta based on the customer's credit terms
+    def _get_delta_qty(self, qty, uom):
+        if uom == 'days':
+            delta_qty = timedelta(days=qty)
+        elif uom == 'weeks':
+            delta_qty = relativedelta(weeks=qty)
+        elif uom == 'months':
+            delta_qty = relativedelta(months=qty)
+        elif uom == 'quarters':
+            delta_qty = relativedelta(months=qty*3)
+        elif uom == 'years':
+            delta_qty = relativedelta(years=qty)
+        return delta_qty
```

Comment:
This doesn't appear to calculate a fiscal quarter properly; it simply looks at the last 3 months instead of determining what the current quarter actually is and gathering invoices based on that value.

If this is the intended function I would recommend removing the 'quarters' option as it could be misleading to users.

E.g. from my original code:
```
...
cur_date = date.today()
cur_quarter = cur_date.month // 3 + 1
valid_invoices = filter(
    lambda invoice:
        (invoice.date_due.month // 3 + 1) ==
        cur_quarter,
        open_invoices
)
```

Additionally, is Swapnesh's recommended approach to generating the relativedelta() parameters not desirable in this case?  I feel that using list comprehension and passing kwargs improves readability.  I know they perform the same ultimate function thought :)

Reply:
@barnarj Your code puts today in the 4th quarter...

I am fine with removing quarter if you are.

Reply:
@max3903 that figures 😂 worked in Auguest, off-by-one error in September.

Quarters are not a requirement of ours, so I am quite okay with removing them if you are.

Thanks!